File: src/main/dart/net/coderextreme/data/indexedfaceset_pixeltexture_plus.dart
// Copyright 2021 by John Carlson.  All rights reserved.
import 'dart:io';
import '../x3d.dart';
var X3D0 =  X3D(

      profile_ : SFString('Interchange'),
      version_ : SFString('4.0'),
      head_ : 
        head(
          meta_ : [
            meta(
              name_ : SFString('title'),
              content_ : SFString('indexedfaceset_pixeltexture_plus.x3d')),

            meta(
              name_ : SFString('Image'),
              content_ : SFString('indexedfaceset_pixeltexture_plus-front.jpg')),

            meta(
              name_ : SFString('Image'),
              content_ : SFString('indexedfaceset_pixeltexture_plus-rear.jpg')),

            meta(
              name_ : SFString('Image'),
              content_ : SFString('indexedfaceset_pixeltexture_plus-top.jpg')),

            meta(
              name_ : SFString('Image'),
              content_ : SFString('indexedfaceset_pixeltexture_plus-bottom.jpg')),

            meta(
              name_ : SFString('Image'),
              content_ : SFString('indexedfaceset_pixeltexture_plus-left.jpg')),

            meta(
              name_ : SFString('Image'),
              content_ : SFString('indexedfaceset_pixeltexture_plus-right.jpg')),

            meta(
              name_ : SFString('reference'),
              content_ : SFString('http://www.nist.gov/vrml.html')),

            meta(
              name_ : SFString('reference'),
              content_ : SFString('http://www.itl.nist.gov/div897/ctg/vrml/vrml.html')),

            meta(
              name_ : SFString('creator'),
              content_ : SFString('http://www.itl.nist.gov/div897/ctg/vrml/members.html')),

            meta(
              name_ : SFString('disclaimer'),
              content_ : SFString('This file was provided by the National Institute of Standards and Technology, and is part of the X3D Conformance Test Suite, available at http://www.nist.gov/vrml.html The information contained within this file is provided for use in establishing conformance to the ISO VRML97 Specification. Conformance to this test does not imply recommendation or endorsement by the National Institute of Standards and Technology. This software can be redistributed and/or modified freely provided that any derivative works bear some notice that they are derived from it, and any modified versions bear some notice that they have been modified.')),

            meta(
              name_ : SFString('info'),
              content_ : SFString('Correct definition and compliance of this conformance scene is maintained by the X3D Working Group, https://www.web3d.org/working-groups/x3d')),

            meta(
              name_ : SFString('translator'),
              content_ : SFString('Michael Kass NIST, Don Brutzman NPS')),

            meta(
              name_ : SFString('translated'),
              content_ : SFString('21 January 2001')),

            meta(
              name_ : SFString('modified'),
              content_ : SFString('13 January 2014')),

            meta(
              name_ : SFString('description'),
              content_ : SFString('Test of browser ability to map a 3 times multiple of an PixelTexture onto an IndexedFaceSet geometry. A 3 by 3 matrix of four equal sized red, green, white and yellow squares in the pixel texture map all the faces of the cube.')),

            meta(
              name_ : SFString('identifier'),
              content_ : SFString('https://www.web3d.org/x3d/content/examples/ConformanceNist/GeometricProperties/TextureCoordinate/indexedfaceset_pixeltexture_plus.x3d')),

            meta(
              name_ : SFString('generator'),
              content_ : SFString('Vrml97ToX3dNist, http://ovrt.nist.gov/v2_x3d.html')),

            meta(
              name_ : SFString('generator'),
              content_ : SFString('X3D-Edit 3.3, https://savage.nps.edu/X3D-Edit'))]),
      Scene_ : 
        Scene(
          children_ : [
            Viewpoint(
              description_ : SFString('Front View')),

            Viewpoint(
              description_ : SFString('Rear View'),
              position_ : SFVec3f([SFDouble(0), SFDouble(0), SFDouble(-10)]),
              orientation_ : SFRotation([SFDouble(0), SFDouble(1), SFDouble(0), SFDouble(3.14)])),

            Viewpoint(
              description_ : SFString('Top View'),
              position_ : SFVec3f([SFDouble(0), SFDouble(10), SFDouble(0)]),
              orientation_ : SFRotation([SFDouble(1), SFDouble(0), SFDouble(0), SFDouble(-1.57)])),

            Viewpoint(
              description_ : SFString('Bottom View'),
              position_ : SFVec3f([SFDouble(0), SFDouble(-10), SFDouble(0)]),
              orientation_ : SFRotation([SFDouble(1), SFDouble(0), SFDouble(0), SFDouble(1.57)])),

            Viewpoint(
              description_ : SFString('Right View'),
              position_ : SFVec3f([SFDouble(10), SFDouble(0), SFDouble(0)]),
              orientation_ : SFRotation([SFDouble(0), SFDouble(1), SFDouble(0), SFDouble(1.57)])),

            Viewpoint(
              description_ : SFString('Left View'),
              position_ : SFVec3f([SFDouble(-10), SFDouble(0), SFDouble(0)]),
              orientation_ : SFRotation([SFDouble(0), SFDouble(1), SFDouble(0), SFDouble(-1.57)])),

            NavigationInfo(
              type_ : MFString([SFString("EXAMINE"), SFString("WALK"), SFString("FLY"), SFString("ANY")])),

            Shape(
              appearance_ : 
                Appearance(
                  material_ : 
                    Material(),
                  texture_ : 
                    PixelTexture(
                      image_ : SFImage([SFInt32(2), SFInt32(2), SFInt32(4), SFInt32(-16776961), SFInt32(16711935), SFInt32(-1), SFInt32(-65281)]))),
              geometry_ : 
                IndexedFaceSet(
                  creaseAngle_ : 0.5,
                  colorPerVertex_ : false,
                  texCoordIndex_ : MFInt32([SFInt32(0), SFInt32(1), SFInt32(3), SFInt32(2), SFInt32(-1), SFInt32(0), SFInt32(1), SFInt32(3), SFInt32(2), SFInt32(-1), SFInt32(0), SFInt32(1), SFInt32(3), SFInt32(2), SFInt32(-1), SFInt32(0), SFInt32(1), SFInt32(3), SFInt32(2), SFInt32(-1), SFInt32(0), SFInt32(1), SFInt32(3), SFInt32(2), SFInt32(-1), SFInt32(0), SFInt32(1), SFInt32(3), SFInt32(2), SFInt32(-1)]),
                  coordIndex_ : MFInt32([SFInt32(0), SFInt32(1), SFInt32(3), SFInt32(2), SFInt32(-1), SFInt32(4), SFInt32(5), SFInt32(7), SFInt32(6), SFInt32(-1), SFInt32(6), SFInt32(7), SFInt32(1), SFInt32(0), SFInt32(-1), SFInt32(2), SFInt32(3), SFInt32(5), SFInt32(4), SFInt32(-1), SFInt32(6), SFInt32(0), SFInt32(2), SFInt32(4), SFInt32(-1), SFInt32(1), SFInt32(7), SFInt32(5), SFInt32(3), SFInt32(-1)]),
                  color_ : 
                    Color(
                      color_ : MFColor([SFColor(0), SFColor(1), SFColor(0), SFColor(0), SFColor(1), SFColor(0), SFColor(0), SFColor(1), SFColor(0), SFColor(0), SFColor(1), SFColor(0), SFColor(0), SFColor(1), SFColor(0), SFColor(0), SFColor(1), SFColor(0)])),
                  texCoord_ : 
                    TextureCoordinate(
                      point_ : MFVec2f([SFVec2f([-1,2]),SFVec2f([-1,-1]),SFVec2f([2,2]),SFVec2f([2,-1])])),
                  coord_ : 
                    Coordinate(
                      point_ : MFVec3f([SFVec3f([-2,1,1]),SFVec3f([-2,-1,1]),SFVec3f([2,1,1]),SFVec3f([2,-1,1]),SFVec3f([2,1,-1]),SFVec3f([2,-1,-1]),SFVec3f([-2,1,-1]),SFVec3f([-2,-1,-1])]))))]));
void main() { exit(0); }

File: src/main/c/net/coderextreme/data/indexedfaceset_pixeltexture_plus.c
#include "C:/x3d-code/www.web3d.org/x3d/languages/c/Concretes.h"
void main(int argc, char ** argv) {
Browser browser = X3D.getBrowser();
X3D X3D0;
X3D0.profile = "Interchange";
X3D0.version = "4.0";
head head1 = createNode("head");
meta meta2 = createNode("meta");
meta2.name = "title";
meta2.content = "indexedfaceset_pixeltexture_plus.x3d";
head1.meta = new MFNode();

head1.meta[0] = meta2;

meta meta3 = createNode("meta");
meta3.name = "Image";
meta3.content = "indexedfaceset_pixeltexture_plus-front.jpg";
head1.meta[1] = meta3;

meta meta4 = createNode("meta");
meta4.name = "Image";
meta4.content = "indexedfaceset_pixeltexture_plus-rear.jpg";
head1.meta[2] = meta4;

meta meta5 = createNode("meta");
meta5.name = "Image";
meta5.content = "indexedfaceset_pixeltexture_plus-top.jpg";
head1.meta[3] = meta5;

meta meta6 = createNode("meta");
meta6.name = "Image";
meta6.content = "indexedfaceset_pixeltexture_plus-bottom.jpg";
head1.meta[4] = meta6;

meta meta7 = createNode("meta");
meta7.name = "Image";
meta7.content = "indexedfaceset_pixeltexture_plus-left.jpg";
head1.meta[5] = meta7;

meta meta8 = createNode("meta");
meta8.name = "Image";
meta8.content = "indexedfaceset_pixeltexture_plus-right.jpg";
head1.meta[6] = meta8;

meta meta9 = createNode("meta");
meta9.name = "reference";
meta9.content = "http://www.nist.gov/vrml.html";
head1.meta[7] = meta9;

meta meta10 = createNode("meta");
meta10.name = "reference";
meta10.content = "http://www.itl.nist.gov/div897/ctg/vrml/vrml.html";
head1.meta[8] = meta10;

meta meta11 = createNode("meta");
meta11.name = "creator";
meta11.content = "http://www.itl.nist.gov/div897/ctg/vrml/members.html";
head1.meta[9] = meta11;

meta meta12 = createNode("meta");
meta12.name = "disclaimer";
meta12.content = "This file was provided by the National Institute of Standards and Technology, and is part of the X3D Conformance Test Suite, available at http://www.nist.gov/vrml.html The information contained within this file is provided for use in establishing conformance to the ISO VRML97 Specification. Conformance to this test does not imply recommendation or endorsement by the National Institute of Standards and Technology. This software can be redistributed and/or modified freely provided that any derivative works bear some notice that they are derived from it, and any modified versions bear some notice that they have been modified.";
head1.meta[10] = meta12;

meta meta13 = createNode("meta");
meta13.name = "info";
meta13.content = "Correct definition and compliance of this conformance scene is maintained by the X3D Working Group, https://www.web3d.org/working-groups/x3d";
head1.meta[11] = meta13;

meta meta14 = createNode("meta");
meta14.name = "translator";
meta14.content = "Michael Kass NIST, Don Brutzman NPS";
head1.meta[12] = meta14;

meta meta15 = createNode("meta");
meta15.name = "translated";
meta15.content = "21 January 2001";
head1.meta[13] = meta15;

meta meta16 = createNode("meta");
meta16.name = "modified";
meta16.content = "13 January 2014";
head1.meta[14] = meta16;

meta meta17 = createNode("meta");
meta17.name = "description";
meta17.content = "Test of browser ability to map a 3 times multiple of an PixelTexture onto an IndexedFaceSet geometry. A 3 by 3 matrix of four equal sized red, green, white and yellow squares in the pixel texture map all the faces of the cube.";
head1.meta[15] = meta17;

meta meta18 = createNode("meta");
meta18.name = "identifier";
meta18.content = "https://www.web3d.org/x3d/content/examples/ConformanceNist/GeometricProperties/TextureCoordinate/indexedfaceset_pixeltexture_plus.x3d";
head1.meta[16] = meta18;

meta meta19 = createNode("meta");
meta19.name = "generator";
meta19.content = "Vrml97ToX3dNist, http://ovrt.nist.gov/v2_x3d.html";
head1.meta[17] = meta19;

meta meta20 = createNode("meta");
meta20.name = "generator";
meta20.content = "X3D-Edit 3.3, https://savage.nps.edu/X3D-Edit";
head1.meta[18] = meta20;

meta meta21 = createNode("meta");
meta21.name = "license";
meta21.content = "../../license.html";
head1.meta[19] = meta21;

head = head1;

Viewpoint Viewpoint23 = createNode("Viewpoint");
Viewpoint23.description = "Front View";
children = new MFNode();

children[0] = Viewpoint23;

Viewpoint Viewpoint24 = createNode("Viewpoint");
Viewpoint24.description = "Rear View";
Viewpoint24.orientation = new SFRotation(new float[0,1,0,3.14]);
Viewpoint24.position = new SFVec3f(new float[0,0,-10]);
children[1] = Viewpoint24;

Viewpoint Viewpoint25 = createNode("Viewpoint");
Viewpoint25.description = "Top View";
Viewpoint25.orientation = new SFRotation(new float[1,0,0,-1.57]);
Viewpoint25.position = new SFVec3f(new float[0,10,0]);
children[2] = Viewpoint25;

Viewpoint Viewpoint26 = createNode("Viewpoint");
Viewpoint26.description = "Bottom View";
Viewpoint26.orientation = new SFRotation(new float[1,0,0,1.57]);
Viewpoint26.position = new SFVec3f(new float[0,-10,0]);
children[3] = Viewpoint26;

Viewpoint Viewpoint27 = createNode("Viewpoint");
Viewpoint27.description = "Right View";
Viewpoint27.orientation = new SFRotation(new float[0,1,0,1.57]);
Viewpoint27.position = new SFVec3f(new float[10,0,0]);
children[4] = Viewpoint27;

Viewpoint Viewpoint28 = createNode("Viewpoint");
Viewpoint28.description = "Left View";
Viewpoint28.orientation = new SFRotation(new float[0,1,0,-1.57]);
Viewpoint28.position = new SFVec3f(new float[-10,0,0]);
children[5] = Viewpoint28;

NavigationInfo NavigationInfo29 = createNode("NavigationInfo");
NavigationInfo29.type = new MFString(new java.lang.String["EXAMINE","WALK","FLY","ANY"]);
children[6] = NavigationInfo29;

Shape Shape30 = createNode("Shape");
Appearance Appearance31 = createNode("Appearance");
Material Material32 = createNode("Material");
Appearance31.material = Material32;

PixelTexture PixelTexture33 = createNode("PixelTexture");
PixelTexture33.image = new SFImage(new int[2,2,4,-16776961,16711935,-1,-65281]);
Appearance31.texture = PixelTexture33;

Shape30.appearance = Appearance31;

IndexedFaceSet IndexedFaceSet34 = createNode("IndexedFaceSet");
IndexedFaceSet34.colorPerVertex = False;
IndexedFaceSet34.coordIndex = new MFInt32(new int[0,1,3,2,-1,4,5,7,6,-1,6,7,1,0,-1,2,3,5,4,-1,6,0,2,4,-1,1,7,5,3,-1]);
IndexedFaceSet34.creaseAngle = 0.5;
IndexedFaceSet34.texCoordIndex = new MFInt32(new int[0,1,3,2,-1,0,1,3,2,-1,0,1,3,2,-1,0,1,3,2,-1,0,1,3,2,-1,0,1,3,2,-1]);
Color Color35 = createNode("Color");
Color35.color = new MFColor(new float[0,1,0,0,1,0,0,1,0,0,1,0,0,1,0,0,1,0]);
IndexedFaceSet34.color = Color35;

Coordinate Coordinate36 = createNode("Coordinate");
Coordinate36.point = new MFVec3f(new float[-2,1,1,-2,-1,1,2,1,1,2,-1,1,2,1,-1,2,-1,-1,-2,1,-1,-2,-1,-1]);
IndexedFaceSet34.coord = Coordinate36;

TextureCoordinate TextureCoordinate37 = createNode("TextureCoordinate");
TextureCoordinate37.point = new MFVec2f(new float[-1,2,-1,-1,2,2,2,-1]);
IndexedFaceSet34.texCoord = TextureCoordinate37;

Shape30.geometry = IndexedFaceSet34;

children[7] = Shape30;

}
